ocaml/runtime/extern.c

```diff
@@ -677,6 +677,10 @@ Caml_inline mlsize_t extern_closure_up_to_env(value v)
     }
   } while (!Is_last_closinfo(info));
   CAMLassert(i <= startenv);
+  /* The non-scanned part of the environment */
+  while (i < startenv) {
+    extern_int(Long_val(Field(v, i++)));
+  }
```

Comment:
OK, I've added the simplest fix I can see for this, which is to introduce a new "unboxed int64" marshalling code.  This is just like `CODE_INT64` but doesn't do the `Val_long` conversion on demarshalling.  The scheme of codes looks tricky, so to avoid disturbing too much, I've reused `CODE_CUSTOM` which has been deprecated now for five years - that seems long enough.  I did wonder about bumping the marshalling format magic number, but it isn't clear to me this is necessary, and I'm unsure what else it would break - so probably best not.

Reply:
(There's no optimization of the new `CODE_UNBOXED_INT64` to a shorter version at present; I don't think this is necessary now since closure marshalling is rare - and we'd like to disable it entirely in the future since it's often unsafe.  When we end up in a world of mixed blocks, we might want to revisit this, but it will require more extensive changes to the marshalling codes.)

Reply:
OK, have rewritten the last commit to do this.